src/components/vendorcomponents/NumberInputWithLabel.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+interface NumberInputProps {
+    label: string;
+    register: any;
+    name: string;
+}
+
+export const NumberInputWithLabel: React.FC<NumberInputProps> = ({
+    label,
+    register,
+    name,
+}) => (
+    <div>
+        <label className="block text-sm font-medium text-gray-700">{label}</label>
+        <input
+            type="number"
+            {...register(name)}
+            className="mt-1 block w-full border border-gray-300 rounded-md shadow-sm p-1"
+        />
+    </div>
```

Comment:
same goes here. you're repeating yourself. the `type` is the only thing that's changing and your making a new component just because of a different input type!